File: Backend/Models/comment.go
package models

type StoryComment struct {
	CommentId       int    `json:"commentId"`
	StoryId         int    `json:"storyId"`
	UserId          int    `json:"userId"`
	ParentCommentId int    `json:"parentCommentId"`
	Content         string `json:"content"`
	DateCreated     string `json:"dateCreated"`
	UpdatedAt       string `json:"updatedAt"`
	Username        string `json:"username"`
	Votes           int    `json:"votes"`
}

File: Frontend/src/app/types/comment.ts
export type StoryComment = {
	commentId: number;
	storyId: number;
	userId: number;
	parentCommentId: number;
	content: string;
	votes: number;
	dateCreated: string;
	updatedAt: string;
	username: string;
};

export type NewStoryComment = {
	storyId: number;
	userId: number;
	parentCommentId: number;
	content: string;
};
